spec/solver_spec.rb

```diff
@@ -0,0 +1,52 @@
+require_relative '../solver'
+
+RSpec.describe Solver do
+  let(:solver) { Solver.new }
+
+  context '#factorial' do
+    it 'should return the right value when given an integer greater than 1' do
+      expect(solver.factorial(3)).to eq(6)
+      expect(solver.factorial(8)).to eq(40_320)
+    end
+
+    it 'should return 1 when the number given is 0' do
+      expect(solver.factorial(0)).to eq(1)
+    end
+
+    it 'should return -1 when the number given is less than 0' do
+      expect(solver.factorial(-5)).to eq(-1)
+    end
+  end
```

Comment:
According to the project requirements, The `#factorial` method should have tests that handle: * Negative value which raises an exception as an expectation 🤓

e.g `expect { resolver.factorial(-5) }.to raise_exception` 👍

Reply:
Fixed.